export_options.go

```diff
@@ -74,23 +100,80 @@ func (b XcodebuildTargetInfoProvider) TargetCodeSignEntitlements(target, configu
 	return b.xcodeProj.TargetCodeSignEntitlements(target, configuration)
 }
 
-// projectEntitlementsByBundleID finds the project's main application target, pointed by the provided scheme and configuration,
+func archivableApplicationTarget(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configurationName string) (*xcodeproj.Target, error) {
+	archiveEntry, ok := scheme.AppBuildActionEntry()
+	if !ok {
+		return nil, fmt.Errorf("archivable entry not found in project: %s for scheme: %s", xcodeProj.Path, scheme.Name)
+	}
+
+	mainTarget, ok := xcodeProj.Proj.Target(archiveEntry.BuildableReference.BlueprintIdentifier)
+	if !ok {
+		return nil, fmt.Errorf("target not found: %s", archiveEntry.BuildableReference.BlueprintIdentifier)
+	}
+
+	return &mainTarget, nil
+}
+
+func dependentExecutableTargetsOf(applicationtarget xcodeproj.Target) (dependentTargets []xcodeproj.Target) {
```

Comment:
Could it be called something like dependentMainAppBundleTargetsOf, meaning like App Clips is also executable right?

Reply:
what abouth `dependentApplicationBundleTargetsOf` ?

Reply:
All right

Reply:
Why filtering App Clips is needed from the dependent targets?

Reply:
Xcode filters it too, i attached an exportOptions generated by Xcode to the JIRA task. I did not test if it works without the filtering.